mantis-common/src/test/java/io/mantisrx/runtime/descriptor/DeploymentStrategyTest.java

```diff
@@ -64,4 +69,33 @@ public void shouldNotRequireInheritInstanceCheck() {
         assertFalse(res.requireInheritInstanceCheck(2));
         assertFalse(res.requireInheritInstanceCheck());
     }
+
+    @Test
+    public void testSerialization() throws Exception {
+        String expected = "{"
+            + "\"stageDeploymentStrategyMap\":"
+            + "{"
+                + "\"1\":{\"inheritInstanceCount\":false},"
+                + "\"2\":{\"inheritInstanceCount\":true},"
+                + "\"3\":{\"inheritInstanceCount\":true}},"
+            + "\"resourceClusterId\":\"rescluster1\""
+            + "}";
+
+        final DeploymentStrategy ds = serializer.fromJSON(expected, DeploymentStrategy.class);
+        assertEquals(expected.replaceAll("[\\n\\s]+", ""), serializer.toJson(ds));
+    }
+
+    @Test
+    public void testSerializationBackCompat() throws Exception {
+        String expected = "{"
+            + "\"stageDeploymentStrategyMap\":"
+            + "{"
+            + "\"1\":{\"inheritInstanceCount\":false},"
+            + "\"2\":{\"inheritInstanceCount\":true},"
+            + "\"3\":{\"inheritInstanceCount\":true}}"
+            + "}";
+
+        final DeploymentStrategy ds = serializer.fromJSON(expected, DeploymentStrategy.class);
+        assertEquals(expected.replaceAll("[\\n\\s]+", ""), serializer.toJson(ds));
```

Comment:
Add a check on what the resourceClusterID is for this JSON.

Reply:
Good point.

Reply:
It's covered in test "testSerialization" above.